### PR TITLE
Delete stale map task interface altering code.

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
If you have a dynamic task that runs a map task, propeller was artificially unnecessarily adding an extra layer of list to the interface.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/2675

### CODE REVIEW

File: pkg/controller/nodes/dynamic/dynamic_workflow.go [hunk 1]
```diff
@@ -101,13 +101,6 @@ func (d dynamicNodeTaskNodeHandler) buildDynamicWorkflowTemplate(ctx context.Con
 			if t.GetContainer() != nil && parentTask.GetContainer() != nil {
 				t.GetContainer().Config = append(t.GetContainer().Config, parentTask.GetContainer().Config...)
 			}
-
-			// TODO: This is a hack since array tasks' interfaces are malformed. Remove after
-			// FlyteKit version that generates the right interfaces is deployed.
-			if t.Type == "container_array" {
-				iface := t.GetInterface()
-				iface.Outputs = makeArrayInterface(iface.Outputs)
```
Comment (hamersaw) [inline pkg/controller/nodes/dynamic/dynamic_workflow.go:109]: Do we want to remove the makeArrayInterface function then as well? It looks like it is not used elsewhere.

Comment (wild-endeavor) [reply pkg/controller/nodes/dynamic/dynamic_workflow.go:109]: yeah just saw that.

**Review by EngHabu — APPROVED** [review]
